Orange/widgets/data/owtransform.py

```diff
@@ -75,27 +75,28 @@ def set_data(self, data):
         self.data = data
         self.set_input_label_text()
 
-    @Inputs.preprocessor
-    def set_preprocessor(self, preprocessor):
-        self.preprocessor = preprocessor
+    @Inputs.template_data
+    @check_sql_input
+    def set_template_data(self, data):
+        self.template_data = data
 
     def handleNewSignals(self):
         self.apply()
 
     def apply(self):
         self.clear_messages()
         self.transformed_data = None
-        if self.data is not None and self.preprocessor is not None:
+        if self.data is not None and self.template_data is not None:
             try:
-                self.transformed_data = self.preprocessor(self.data)
-            except Exception as ex:   # pylint: disable=broad-except
-                self.Error.pp_error(ex)
+                self.transformed_data = self.data.transform(
+                    self.template_data.domain)
```

Comment:
`transformed_data` is stored just for report? Consider pre-preparing (that part of) report and storing it instead.

Reply:
It is actually used many times (for setting labels, reporting, merging data). 
Of course, I could prepare all needed properties, but I'd rather not. It would make the code less clear, and I'd would have to save something anyway.

Reply:
Using instance attributes to pass (temporary-ish) data between methods is like using global variables to pass data between functions. Attributes are meant for storing data that needs to be *stored*. (Though practicality beats purity.)

Please feel free to remind me of that when *I* am doing it. :)